driver-core/src/main/java/com/datastax/driver/core/DefaultResultSetFuture.java

```diff
@@ -156,7 +156,7 @@ public void run() {
                     setException(new DriverInternalError(String.format("Got unexpected %s response from %s", response.type, connection.address)));
                     break;
             }
-        } catch (RuntimeException e) {
+        } catch (Throwable e) {
```

Comment:
Apart from catching ```AssertionError``` (in case of my test) this might be useful with identifying ``` OutOfMemoryError``` or ```StackOverflowError```. Otherwise you might end up with all of application threads been blocked waiting for futures to resolve, as @olim7t wrote. And thread dump won't help you find the cause of the problem.

Of course, there might be a better solution, so maybe it is worth to revert this change and create a new issue to address the problem.